src/devices/lucky_transducer.c

```diff
@@ -0,0 +1,167 @@
+/** @file
+    LUCKY Wireless Transducer - only compatible with Lucky Fishfinder FF718LiC-W, FF1108-1CW, FFW718.
+
+    Copyright (C) 2022 Andre Lohmann <lohmann.andre@gmail.com>
+
+    With the inspiration of Haras1111, jurs, DrDiettrich, agmue and FriedChips from the following thread:
+    https://forum.arduino.cc/t/funksignal-decodieren-fur-masterarbeit-fishfinder-zweckentfremden/439091
+    Whoever you are, kudos to you
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/**
+LUCKY Wireless Transducer - Wireless Sonar Transducer
+
+    https://www.luckysonar.com/wp-content/uploads/2021/11/FF1108-W.pdf
+
+!!! FUnctions are copied from vaillant_vrt340f.c !!!
+
+The data is sent differential Manchester encoded
+with bit-stuffing (after five 1 bits an extra 0 bit is inserted)
+
+All bytes are sent with least significant bit FIRST (1000 0111 = 0xE1)
+
+    0x00 00 7E | 6D F6 | 00 20 00 | 00 | 80 | B4 | 00 | FD 49 | FF 00
+      SYNC+HD. | DevID | CONST?   |Rep.|Wtr.|Htg.|Btr.|Checksm| EPILOGUE
+
+- CONST? ... Unknown, but constant in all observed signals
+- Rep.   ... Repeat indicator: 0x00=original signal, 0x01=first repeat
+- Wtr.   ... pre-heated Water: 0x80=ON, 0x88=OFF (bit 8 is always set)
+- Htg.   ... Heating: 0x00=OFF, 0xB4=ON (2-point), 0x01-0x7F=target heating water temp
+             (bit 8 indicates 2-point heating mode, bits 1-7 the heating water temp)
+- Btr.   ... Battery: 0x00=OK, 0x01=LOW
+- Checksm... Checksum (2-byte signed int): = -sum(bytes 4-12)
+
+*/
+
+#include "decoder.h"
+
+static int validate_checksum(r_device *decoder, uint8_t *b, int from, int to, int cs_from, int cs_to)
```

Comment:
unroll this function to the line where it's used. don't use functions i you avoid it.